### PR TITLE
Staging and Production Enhancement in at_onboarding_widget

### PR DESCRIPTION
**Pull request by yahu1031**
## What I did ##
Added an enhancement for production and staging environment for Onboarding widget.

## Usage ##

Staging usage

```diff
Onboarding(
  context: context,
  atClientPreference: atClientPrefernce,
- domain: AppConstants.rootDomain,
  appColor: const Color.fromARGB(255, 240, 94, 62),
  onboard: (Map<String?, AtClientService> value, String? atsign) {
    AtService.getInstance().atClientServiceMap = value;
    _logger.finer('Successfully onboarded $atsign');
  },
  onError: (Object? error) {
    _logger.severe('Onboarding throws $error error');
  },
  nextScreen: DashBoard(),
+ rootEnvironment: 'staging',
- appAPIKey: AppConstants.devAPIKey,
);
```

Production usage

```diff
Onboarding(
  context: context,
  atClientPreference: atClientPrefernce,
- domain: AppConstants.rootDomain,
  appColor: const Color.fromARGB(255, 240, 94, 62),
  onboard: (Map<String?, AtClientService> value, String? atsign) {
    AtService.getInstance().atClientServiceMap = value;
    _logger.finer('Successfully onboarded $atsign');
  },
  onError: (Object? error) {
    _logger.severe('Onboarding throws $error error');
  },
  nextScreen: DashBoard(),
+ rootEnvironment: 'production',
+ appAPIKey: YOUR_API_KEY,
);
```

## Description for the changelog ##

- This may lead to breaking.
- Production must need an API key where staging doesn't need any.
- Fix for #167

### CODE REVIEW

**Review by charannnreddy — APPROVED** [review]

File: at_onboarding_flutter/example/lib/utils/app_constants.dart [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class AppConstants {
-  static String rootDomain = 'root.atsign.org';
+  static String rootDomain = 'root.atsign.wtf';
```
Comment (murali-shris) [inline at_onboarding_flutter/example/lib/utils/app_constants.dart:4]: this cannot  be wtf in trunk

File: at_onboarding_flutter/lib/screens/pair_atsign.dart [hunk 2]
```diff
@@ -262,8 +261,11 @@ class _PairAtsignWidgetState extends State<PairAtsignWidget> {
         loading = true;
       });
       for (PlatformFile pickedFile in result?.files ?? <PlatformFile>[]) {
-        String? path = pickedFile.path.toString();
-        File selectedFile = File(path.toString());
+        String? path = pickedFile.path;
+        if (path == null) {
+          throw const FileSystemException('FilePicker.pickFiles returned a null path');
+        }
+        File selectedFile = File(path);
```
Comment (sarika01) [inline at_onboarding_flutter/lib/screens/pair_atsign.dart:268]: File does not accept null and path is a nullable string.

File: at_onboarding_flutter/example/lib/main.dart [hunk 3]
```diff
@@ -37,36 +37,36 @@ class _MyAppState extends State<MyApp> {
...
+                        atClientPreference: atClientPrefernce,
+                        appColor: const Color.fromARGB(255, 240, 94, 62),
+                        onboard: (Map<String?, AtClientService> value, String? atsign) {
+                          AtService.getInstance().atClientServiceMap = value;
+                          _logger.finer('Successfully onboarded $atsign');
+                        },
+                        onError: (Object? error) {
+                          _logger.severe('Onboarding throws $error error');
+                        },
+                        rootEnvironment: RootEnvironment.Production,
```
Comment (sarika01) [inline at_onboarding_flutter/example/lib/main.dart]: If you are using RootEnvironment.Production as example then APIKey has to be provided.

File: at_onboarding_flutter/lib/screens/pair_atsign.dart (hunk 2 above)
Comment (yahu1031) [reply at_onboarding_flutter/lib/screens/pair_atsign.dart:268]: Nope if the path is null it won't go to the next step. `File(path)` will be non-nullable value as `path == null` doesn't pass.

File: at_onboarding_flutter/example/lib/main.dart (hunk 3 above)
Comment (yahu1031) [reply at_onboarding_flutter/example/lib/main.dart]: oops i forgot lemme add it to `staging`

File: at_onboarding_flutter/example/lib/utils/app_constants.dart (hunk 1 above)
Comment (sarika01) [reply at_onboarding_flutter/example/lib/utils/app_constants.dart:4]: This is only for the example I suppose. If production endpoint is specified then API key will have to be provided.

**Review by sarika01 — APPROVED** [review]

**Pull request merged**